src/debug/telemetry/telemetry.c

```diff
@@ -22,12 +23,20 @@
 
 LOG_MODULE_DECLARE(ipc, CONFIG_SOF_LOG_LEVEL);
 
+#define PERFORMANCE_DATA_ENTRIES_COUNT (CONFIG_MEMORY_WIN_3_SIZE / sizeof(struct perf_data_item_comp))
+
+SYS_BITARRAY_DEFINE_STATIC(performance_data_bit_array, PERFORMANCE_DATA_ENTRIES_COUNT);
+
 struct perf_bitmap {
 	sys_bitarray_t *array;
 	uint16_t occupied;
 	size_t size;
 };
 
+struct perf_bitmap performance_data_bitmap;
+
+volatile struct perf_data_item_comp *perf_data_;
```

Comment:
`volatile`? Maybe `static` - for as many of them as possible? And the terminating underscore